### ISSUE TITLE
Amazon Games not working since update

### ISSUE THREAD

**Issue by hielfx** [moraroy/NonSteamLaunchers-On-Steam-Deck#486]
Hi, I recently updated the script and launchers but I cannot get Amazon Games to work.

I'm using the following:
- Script version v3.9.7
- Proton version GE-Proton9-18
- Decky plugin v3.0.4 in testing mode
- Decky NonSteamLaunchers plugin v1.0.1
- Steam deck with beta updates

I've tried uninstalling and installing again, moving it to SD, leaving it with th default prefix, using app prefix, installing with both Decky plugin and desktop shortcut, launching it from desktop and game mode, trying several proton versions. Nothing seems to work.

What happens is the app seems like it never starts. With the other launchers (in my case Epic Games and GoG) happens the same but I can resume the game (for some reason) after pressing steam button after a short while and it will display the interface.

**Comment by moraroy** [comment]
@hielfx hey man thank you for finding this out, I had no idea since mine was still installed. It seems it's broken from ge 17 and 18, using 16 seems to install correctly but in a new prefix :/ let me try and do more testing and see what's going on, may take a bit.

**Comment by moraroy** [comment]
@hielfx Amazon games and gog seem to be broken now :( proton ge 17-20 don't seem to work like they used to. I think something has changed recently and I don't want to mess with people's prefixes too much, Amazon games isn't calling a .dll like it used to anymore. The .dll it needs is in fact in the prefix but the "function" its trying to call is broken. I'm assuming this means a dependency is out of date. Not sure how to fix just yet. I'll reply back with the exact .dll a bit later on.

**Comment by pratt-jeremy** [comment]
Just ran through and GoG is working from a fresh start.  Amazon games isn't.  This is from loading the plugin version from desktop mode into Decky and then running the install scripts through the plugin.

**Comment by AoS-DFA** [comment]
Is this issue still active?  Amazon Games did not install when I tried with version 3.9.8.  It didn't throw off any error messages, it just doesn't appear in the library after the install finishes.

Thanks!

**Comment by moraroy** [comment]
@AoS-DFA it should, it installed for me do you have any logs?

**Comment by AoS-DFA** [comment]
Uploaded latest log file.  It looks like the script thinks Amazon Games is already installed.  The script seems to have a problem completely deleting old installers.  I've tried manually deleting everything, and still get this error.
[NonSteamLaunchers-install.log](https://github.com/user-attachments/files/18224404/NonSteamLaunchers-install.log)

Thanks for your help!

**Comment by moraroy** [comment]
@AoS-DFA i seee, thank you for the log, will continue to look at the issue

**Comment by Karrq** [comment]
I had similar problem after updating the launcher - switched to GE Proton 9-22 and I could open the launcher again

**Comment by moraroy** [comment]
@Karrq it's not working with the latest version of ge?

**Comment by Karrq** [comment]
GE Proton 9-27 also works to open the launcher. Nevertheless, games don’t launch directly thru the shortcut, but work thru the launcher

**Comment by moraroy** [comment]
@Karrq which games?  I have some Amazon on mine currently and they can be launched. Are they umu shortcuts?

**Comment by Karrq** [comment]
Yes they are umu shortcuts. I was trying to launch cultist simulator and neurovoided. I can try others as well, I have apico, algobot and titan souls installed, plus others I can install to try out if you want

**Comment by moraroy** [comment]
@Karrq yeah :/ it's because of the bug in the desktop version. Basically, I've been trying to fix it for months, the compat tool keeps getting "checked" so for umu shortcuts with NSL uncheck the force compatability.

**Comment by Karrq** [comment]
Oh I did notice that the compatibility tools was unset, and that it wasn't working, I thought actually that UMU would take care of it, but I also tried setting the compat tool the same as what I have on the launcher (using separate prefix feature) but still no luck.
Basically can't launch amazon games with or without GE Proton 9-27 set on UMU shortcuts

**Comment by moraroy** [comment]
@Karrq look inside the launch options of the umu shortcut what proton is it trying to use? There is no logic yet for nsl to remake shortcuts based on new proton gets you would have to change the number based on what you have

**Comment by Karrq** [comment]
Oh, I see, I need to change `PROTONPATH` in the launch options then... Couldn't NSL set it to the shortcut properties instead? Since we need to update it manually anyways...

The shortcut was still using 9-20, which was the version I was using originally that stopped working.
I tried both updating `PROTONPATH` to 9-27 as well as removing the env variable altogether and setting it in the shortcut options instead (as compat tool), but both led to the same behaviour: shortcut was remade, and the game would initially load but fail to launch shortly after.

**Comment by moraroy** [comment]
@Karrq I was trying to do that originally it would be better for me to "force it" yeah when nsl remakes it it should give you a proper launch option, and then just make sure it's not checked and make sure you have that proton installed. Should work, if not I'll try to look at it this weekend.

**Comment by Karrq** [comment]
When the shortcut was remade I still got 9-20, but regardless using the manually edited shortcut still failed to launch

**Comment by moraroy** [comment]
@Karrq you probably don't have umu installed, a while back I had fix a bug, try to use the button on nsl called "Update Proton -Ge" this will also install umu properly.

**Comment by Karrq** [comment]
Ok after pressing the update button I got the updated shortcuts, but still, game fails to launch thru the shortcut but works with launcher. Maybe amazon games changes their CLI flags?

**Comment by moraroy** [comment]
@Karrq my amazon games are being called fine though, i have a game called ape out. check for umu.. /home/deck/bin/there should be two files in there make sure they are set to executable, once you check that and everything is good, make sure the the game paths make sense in the shortcuts, and make sure the umu shortcuts are unchecked. other than that there isnt too much to it....i would bet that some where down the line that there is a wrong path somewhere, not by you, but possible my script

**Comment by Karrq** [comment]
I tried updating again a couple of times, eventually it worked - I guess UMU wasn't installed properly before, but it is now.
So, to recap, GE 9-27 works for launcher & games shortcut after update.

If UMU is not installed the symptoms vary from: launch fails instantly, launch fails after a few seconds.
If Proton is the wrong version: launch fails after 1 second.
